src/state.rs

```diff
@@ -562,14 +562,20 @@ impl DemandProxyState {
             );
             return Err(Error::NoValidDestination(Box::new(dst_workload.clone())));
         }
-        let ip = Box::pin(self.resolve_workload_address(dst_workload, src_workload)).await?;
+        let ip = Box::pin(self.resolve_workload_address(
+            dst_workload,
+            src_workload,
+            original_target_address,
```

Comment:
all we really care about is whether the original_target_address is ipv6 or not. Do we need to pass around a SocketAddr for that or would it be better to simply pass around a bool or enum to represent the preferred ip family?

Reply:
Technically no, but I was thinking it is a bit more flexible and conveys the meaning of the variable well. We change between whether we need the IP, the port, the family, etc a few times in this codepath (and just above we use the full IP for instance)